shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -421,14 +421,16 @@ final class ImageReaderSurfaceProducer
     // If we cleanup the ImageReaders on memory pressure it breaks VirtualDisplay
     // backed platform views. Disable for now as this is only necessary to work
     // around a Samsung-specific Android 14 bug.
-    private static final boolean CLEANUP_ON_MEMORY_PRESSURE = false;
+    private static final boolean CLEANUP_ON_MEMORY_PRESSURE = true;
 
     private final long id;
 
     private boolean released;
     // Will be true in tests and on Android API < 33.
     private boolean ignoringFence = false;
 
+    private boolean trimOnMemoryPressure = CLEANUP_ON_MEMORY_PRESSURE;
```

Comment:
@johnmccutchan if you set this to visible for testing and then override the failing test to set `trimOnMemoryPressure` to false then the tests passes. I still think this pr should add tests for the other behavior but that would be the fastest way to land this pr.

Reply:
I think it'll be way more clear to just add the missing tests instead of making this variable visible for testing.

Reply:
Tests are coming.

Reply:
To verify both sides of this boolean we would have to make this variable public and since I don't think it should be part of the API visible for testing makes sense.

Reply:
Extended the tests.